lib/util/SortableSet.js

```diff
@@ -1,17 +1,30 @@
 "use strict";
 
+/**
+ * A subset of Set that offers sorting functionality
+ * @template T
+ * @typedef {(a: T, b: T) => number} SortFn
+ */
 class SortableSet extends Set {
+	/**
+	 * Create a new sortable set
+	 * @param {IterableIterator<T>=} initialIterable
+	 * @param {SortFn=} defaultSort
+	 */
 	constructor(initialIterable, defaultSort) {
 		super(initialIterable);
+		/** @private @type {SortFn} */
 		this._sortFn = defaultSort;
+		/** @private @type {SortFn | null} */
 		this._lastActiveSortFn = null;
+		/** @private @type {Map<Function, T> | undefined} */
```

Comment:
`Map<Function, any>`

Reply:
`Set` is strict about types. Wouldn't it make sense to make `SortableSet` strict too? `T` can default to `any` but if `T` is provided by the generic syntax I think we should enforce it.

This is ok:

```ts
const s = new Set()

s.add(1)
s.add('')
```

This is not ok:

```ts
const s = new Set<number>()

s.add(1)
s.add('')
```